charms/worker/k8s/src/charm.py

```diff
@@ -383,39 +383,62 @@ def _get_valid_annotations(self) -> Optional[dict]:
 
         return annotations
 
-    def _configure_annotations(self, config: BootstrapConfig):
-        """Configure the annotations for the Canonical Kubernetes cluster.
+    def _get_cluster_config_from_charm_config(self) -> UserFacingClusterConfig:
+        """Retrieve the cluster config from charm configuration.
+
+        Returns:
+            UserFacingClusterConfig: The expected cluster configuration.
+        """
+        local_storage = LocalStorageConfig(
```

Comment:
I think we need to add `allow_population_by_field_name = True` to the `LocalStorageConfigModel`. Otherwise _I think_ the fields with aliases (e.g. `local_path`) are not going to get populated like this: https://programiz.pro/ide/python/ORE3ZZSJRG?utm_medium=playground&utm_source=python_playground-shared-project-link
```python
class LocalStorageConfig(BaseModel):
    """Configuration for the local storage settings of the cluster.

    Attributes:
        enabled: Optional flag which represents the status of Storage.
        local_path: The local path for storage.
        reclaim_policy: The policy for reclaiming local storage.
        set_default: Optional flag to set this as the default storage option.
    """

    enabled: Optional[bool] = Field(None)
    local_path: Optional[str] = Field(None, alias="local-path")
    reclaim_policy: Optional[str] = Field(None, alias="reclaim-policy")
    set_default: Optional[bool] = Field(None, alias="set-default")

    class Config:
        """Pydantic configuration for the LocalStorageConfig model."""
        allow_population_by_field_name = True
```

p.s. Wish we had an easy Python playground just like Go...

Reply:
[GIVEN THAT I'M CORRECT] I found this accidentally trying to understand why is Pylance going crazy and underlining everything in red. 
Maybe we need to add a test that populates and validates this config so that we don't miss anything.

Reply:
Thanks - for some reason it worked anyway (maybe the settings from `UserfacingConfig` are applied recursively?

Anyway, added it to be explicit.